File: basics/Q23/main.go
package main

import "fmt"

func main() {
	var (
		a4  int
		pen int
		t   float64
	)
	fmt.Print("Enter A4 price: ")
	fmt.Scan(&a4)
	fmt.Print("Enter pen price: ")
	fmt.Scan(&pen)
	fmt.Print("Enter t: ")
	fmt.Scan(&t)

	cost := 50*float64(pen)*t/100 + 150*float64(a4)*t/100
	fmt.Println("Extra cost is", cost)
}

File: basics/Q23/main.rs
use std::io;
use std::str::FromStr;

fn main() {
    let a: i32 = read_input("Enter A4 price:");
    let pen: i32 = read_input("Enter pen price:");
    let t: f64 = read_input("Enter t:");
    let cost = 50.0 * pen as f64 * t / 100.0 + 150.0 * a as f64 * t / 100.0;
    println!("Extra cost is {}", cost)
}

fn read_input<T: FromStr>(prompt: &str) -> T {
    println!("{}", prompt);

    let mut input = String::new();
    io::stdin()
        .read_line(&mut input)
        .expect("Failed to read line");

    match input.trim().parse::<T>() {
        Ok(value) => return value,
        Err(_e) => panic!("Invalid input: please try again."),
    }
}
